backend/src/Controllers/PostsImportController.php

```
@@ -0,0 +1,337 @@
+					$post_data    = [
+						'comment_status' => $property['WP:COMMENT_STATUS'],
+						'ping_status'    => $property['WP:PING_STATUS'],
+						'post_author'    => $current_user->ID,
+						'post_content'   => $property['DESCRIPTION'] ? $property['DESCRIPTION'] : '',
+						'post_excerpt'   => $property['EXCERPT:ENCODED'] ? $property['EXCERPT:ENCODED'] : '',
+						'post_name'      => $property['WP:POST_NAME'],
+						'post_status'    => $property['WP:STATUS'],
+						/* translators: %s: post/page title */
+						'post_title'     => $property['TITLE'],
+						'post_type'      => $property['WP:POST_TYPE'],
+						'to_ping'        => $property['WP:PING_STATUS'],
+						'menu_order'     => $property['WP:MENU_ORDER'],
+					];
+
+					if ( ! function_exists( 'post_exists' ) ) {
+						require_once ABSPATH . 'wp-admin/includes/post.php';
+					}
+					if ( 0 === post_exists( $property['TITLE'], '', '', $property['WP:POST_TYPE'] ) ) {
+
+						/* Import Attachments */
+
+						if ( $property['WP:POST_TYPE'] == 'attachment' ) {
+
+							$file = $property['WP:ATTACHMENT_URL'];
+							$filename = basename( $file );
+							$read_file = file_get_contents( $file );
+
+							if ( $read_file ) {
+
+								$upload_file = wp_upload_bits( $filename, null, $read_file );
+
+								if ( ! $upload_file['error'] ) {
+
+									$wp_filetype = wp_check_filetype( $filename, null );
+									$post_data['post_mime_type'] = $wp_filetype['type'];
+									$attachment_id = wp_insert_attachment( $post_data, $upload_file['file'] );
+									if ( ! is_wp_error( $attachment_id ) ) {
+										require_once( ABSPATH . 'wp-admin' . '/includes/image.php' );
+										$attachment_data = wp_generate_attachment_metadata( $attachment_id, $upload_file['file'] );
+										wp_update_attachment_metadata( $attachment_id, $attachment_data );
+									}
+
+									$sucess_imp_count++;
+
+								} else {
+									$fail_imp_count++;
+									return rest_ensure_response(
```

Do we really want to cancel the entire import if an attachment fails? It might be better to fail silently for attachments and let the import proceed.